src/routes/launchpad/+page.svelte

```diff
@@ -0,0 +1,676 @@
+		minCommitment.style.left = `${barWidth / 2}px`;
+	};
+
+	const setUpSns = async () => {
+		const agent = HttpAgent.createSync({
+			host: 'http://127.0.1:8080'
+		});
+
+		if (DEV) {
+			agent.fetchRootKey();
+		}
+
+		const CANISTER_ID = 'bd3sg-teaaa-aaaaa-qaaba-cai';
+
+		snsCanister = Actor.createActor(idlFactorySns, {
+			agent,
+			canisterId: CANISTER_ID
+		});
+	};
+
+	const claimWtn = async () => {
+		isNotAvailable = true;
+		try {
+			const result = await snsCanister.claim_wtn(Principal.fromText(participant));
+			const key = Object.keys(result)[0];
+			switch (key) {
+				case 'Err':
+					toasts.add(Toast.error((result as { Err: string }).Err));
+					break;
+				case 'Ok':
+					toasts.add(
+						Toast.success(`Successful transfer at block ${(result as { Ok: bigint }).Ok}`)
+					);
+			}
+		} catch (e) {
+			console.error(e);
+		}
+		isNotAvailable = false;
+	};
+
+	const updateBalance = async () => {
+		if (!destination || !$canisters) return;
+		try {
+			const { e8s } = await $canisters.icpLedger.anonymousActor.account_balance({
+				account: AccountIdentifier.fromHex(destination).toUint8Array()
+			});
+			balance = e8s;
+		} catch (e) {
+			console.log(e);
+		}
+	};
+
+	const fetchStatus = async () => {
+		try {
+			status = await snsCanister.get_status();
+			resetTimerToElapsedTime(status.time_left / 1_000_000_000n);
+			startTimer();
+		} catch (e) {
+			console.log(e);
+		}
+	};
+
+	const setDestination = async () => {
+		isNotAvailable = true;
+		try {
+			destination = await snsCanister.get_icp_deposit_address(Principal.fromText(participant));
+			await updateBalance();
+			await updateWtnAllocation();
+		} catch (e) {
+			console.log(e);
+		}
+		isNotAvailable = false;
+	};
+
+	const notifyDeposit = async () => {
+		isNotAvailable = true;
+		try {
+			const result = await snsCanister.notify_icp_deposit(Principal.fromText(participant), balance);
+			const key = Object.keys(result)[0];
+			switch (key) {
+				case 'Err':
+					toasts.add(Toast.error((result as { Err: string }).Err));
+					break;
+				case 'Ok':
+					toasts.add(
+						Toast.success(`Successful transfer at block ${(result as { Ok: bigint }).Ok}`)
+					);
+			}
+		} catch (e) {
+			console.log(e);
+		}
+		isNotAvailable = false;
+	};
+
+	onMount(() => {
+		setUpSns().then(async () => {
+			fetchStatus();
+			setPositions();
+		});
+
+		const intervalId = setInterval(async () => {
+			await fetchStatus();
+			await updateBalance();
+			await updateWtnAllocation();
+			setPositions();
+		}, 5000);
+
+		return () => clearInterval(intervalId);
+	});
+
+	$: status;
+</script>
+
+<main>
+	<div class="header-container">
+		<p>
+			WaterNeuron is a liquid staking protocol designed for the Internet Computer network. Staking
+			ICP becomes straightforward and efficient.
+		</p>
+		<span>https://docs.waterneuron.fi | https://x.com/waterneuron | https://t.me/waterneuron</span>
```

Comment:
If you want to put links, please use a tag.

Reply:
done